Neos.Neos/Resources/Private/Fusion/Prototypes/Shortcut.fusion

```diff
@@ -1,15 +1,132 @@
 # Neos.Neos.Shortcut is given a representation for editing purposes
 #
-prototype(Neos.Neos:Shortcut) < prototype(Neos.Fusion:Template) {
-  templatePath = 'resource://Neos.Neos/Private/Templates/FusionObjects/Shortcut.html'
-
+prototype(Neos.Neos:Shortcut) < prototype(Neos.Fusion:Component) {
   targetMode = ${q(node).property('targetMode')}
   firstChildNode = ${q(node).children('[instanceof Neos.Neos:Document]').get(0)}
   target = ${q(node).property('target')}
   targetConverted = ${Neos.Link.hasSupportedScheme(this.target) ? Neos.Link.convertUriToObject(this.target, node) : null}
   targetSchema = ${Neos.Link.getScheme(this.target)}
 
-  node = ${node}
+  i18n = ${q(Neos.BackendUser.get()).property('preferences.interfaceLanguage')}
+
+  renderer = Neos.Fusion:Match {
+    @subject = ${props.targetMode}
+  }
+
+  // targetMode == 'selectedTarget'
+  renderer.selectedTarget = Neos.Fusion:Join {
+    generalMessage = Neos.Fusion:Renderer {
+      fallback = 'This is a shortcut to a specific target:'
+      renderer = ${I18n.id('shortcut.toSpecificTarget').package('Neos.Neos').source('Main').value(this.fallback).locale(props.i18n).translate()}
+    }
+    ifTarget = Neos.Fusion:Case {
+      default {
+        @position = 'end'
+        condition = true
+        renderer = Neos.Fusion:Renderer {
+          fallback = 'no target has been selected'
+          renderer = ${I18n.id('shortcut.noTargetSelected').package('Neos.Neos').source('Main').value(this.fallback).locale(props.i18n).translate()}
+        }
+      }
+    }
+    @glue = "<br/>"
+  }
+
+  renderer.selectedTarget.ifTarget.condition {
+    condition = ${props.target}
+    renderer = Neos.Fusion:Match {
+      @subject = ${props.targetSchema}
+
+      node = Neos.Fusion:Renderer {
+        targetUriTag = Neos.Neos:NodeLink {
+          node = ${props.targetConverted}
+        }
+        fallback = afx`
+          Click {this.targetUriTag} to continue to the page.
+        `
+        renderer = ${I18n.id('shortcut.clickToContinueToPage').package('Neos.Neos').source('Main').arguments([this.targetUriTag]).value(this.fallback).locale(props.i18n).translate()}
```

Comment:
Yeah these lines are a bit long and mostly the same (besides the id). They use `this.targetUriTag` and `this.fallback`.

Should this be implemented a bit cleaner? With a custom EEL helper f.x.?

Reply:
Solved thanks to https://mind-the-seb.de/de/blog/der-translation-helper-in-neos-cms